File: c++/2812.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

vector<char> stack;

int main() {
  freopen("example.txt", "r", stdin);

  int N, K;
  cin >> N >> K;

  string s;
  cin >> s;

  for (int i = 0; i < s.length(); i++) {
    if (stack.size() == 0)
      stack.push_back(s[i]);
    else {
      while (K != 0 && stack.size() != 0 && stack.back() < s[i]) {
        stack.pop_back();
        K--;
      }
      stack.push_back(s[i]);
    }
  }

  while (K--) {
    stack.pop_back();
  }

  for (int i = 0; i < stack.size(); i++) {
    cout << stack[i];
  }

  return 0;
}

File: js/2812.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

let [N, K] = inputs[0].split(' ').map(Number);
const s = inputs[1];

const stack = [];

for (let i = 0; i < s.length; i++) {
  if (stack.length === 0) stack.push(s[i]);
  else {
    while (K !== 0 && stack[stack.length - 1] < s[i]) {
      stack.pop();
      K--;
    }
    stack.push(s[i]);
  }
}

console.log(K);

while (K !== 0) {
  stack.pop();
  K--;
}

console.log(stack.join(''));
